includes/class-newspack-blocks.php

```diff
@@ -1389,6 +1389,22 @@ public static function get_displayed_post_date( $post = null ) {
 		return apply_filters( 'newspack_blocks_displayed_post_date', mysql_to_rfc3339( $post->post_date ), $post );
 	}
 
+	/**
+	 * Get post date in ISO-8601 format to be used in the datetime attribute.
+	 *
+	 * @param WP_Post $post Post object.
+	 * @return string Date string in ISO-8601 format.
+	 */
+	public static function get_datetime_post_date( $post = null ) {
+		if ( $post === null ) {
+			$post = get_post();
+		}
+		$date  = self::get_displayed_post_date( $post );
+		$date  = new DateTime( $date );
+		$date_formatted = date_i18n( 'c', $date->getTimestamp() );
+		return apply_filters( 'newspack_blocks_displayed_post_date', $date_formatted, $post );
```

Comment:
```suggestion
return apply_filters( 'newspack_blocks_displayed_post_date', get_post_datetime( $post )->format( 'c' ), $post );
```

Because a datetime is never localized or should be changed in any way (other than to change the actual date used), I think we can go simpler and just format the date. What do you think?

Reply:
Ah good call @naxoc -- thank you! I definitely overdid it 😄 This has been updated in [a097c66](https://github.com/Automattic/newspack-blocks/pull/2113/commits/a097c66a522ebc614bab9378ce0020c6ce8f8a8b).